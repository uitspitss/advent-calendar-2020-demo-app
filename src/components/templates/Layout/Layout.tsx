import React, { FC, ReactNode } from 'react';
import { styled } from 'twin.macro';

import { Header } from 'components/organisms/Header';
import { Footer } from 'components/organisms/Footer';
import { CookieNotification } from 'components/molecules/CookieNotification';

export type LayoutProps = {
  children: ReactNode;
  className?: string;
};

type Props = LayoutProps;

const Component: FC<Props> = (props) => {
  const { children, className } = props;

  return (
    <div className={className}>
      <CookieNotification className="cookieNotification" />
      <Header />
      <main>{children}</main>
      <Footer />
    </div>
  );
};

const StyledComponent = styled(Component)``;

/**
 * レイアウト
 */
export const Layout: FC<LayoutProps> = (props) => (
  <StyledComponent {...props} />
);
