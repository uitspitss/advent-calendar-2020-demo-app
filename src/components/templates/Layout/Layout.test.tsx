import React from 'react';
import renderer from 'react-test-renderer';

import { Layout } from './Layout';

describe('Layout', () => {
  test('Snap Shot', () => {
    const tree = renderer
      .create(
        <Layout>
          <a href="/">link</a>
        </Layout>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
