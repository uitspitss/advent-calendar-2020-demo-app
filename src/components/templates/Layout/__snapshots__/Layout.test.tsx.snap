// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Snap Shot 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateY(-100%);
    -ms-transform: translateY(-100%);
    transform: translateY(-100%);
  }

  to {
    -webkit-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }
}

.emotion-0 {
  -webkit-animation: animation-0 0.5s ease;
  animation: animation-0 0.5s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --bg-opacity: 1;
  background-color: rgba(66,153,225,var(--bg-opacity));
  --text-opacity: 1;
  color: rgba(255,255,255,var(--text-opacity));
  font-size: 0.875rem;
  font-weight: 700;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
}

.emotion-0 > div > .acceptButton {
  --bg-opacity: 1;
  background-color: rgba(237,100,166,var(--bg-opacity));
  --text-opacity: 1;
  color: rgba(255,255,255,var(--text-opacity));
  font-weight: 700;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  margin-right: 0.5rem;
  border-style: none;
  cursor: pointer;
}

.emotion-0 > div > .acceptButton:hover {
  --bg-opacity: 1;
  background-color: rgba(184,50,128,var(--bg-opacity));
}

.emotion-0 > div > .declineButton {
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  --text-opacity: 1;
  color: rgba(255,255,255,var(--text-opacity));
  font-weight: 700;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-style: none;
  cursor: pointer;
}

.emotion-0 > div > .declineButton:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6 > div {
  --bg-opacity: 1;
  background-color: rgba(163,191,250,var(--bg-opacity));
  margin-bottom: 0.5rem;
}

@media (min-width:1024px) {
  .emotion-6 > div {
    margin-bottom: 1rem;
  }
}

.emotion-6 > div > nav {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width:1024px) {
  .emotion-6 > div > nav {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

.emotion-6 > div > nav > .title {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.5rem;
}

.emotion-6 > div > nav > .title > span {
  --text-opacity: 1;
  color: rgba(247,250,252,var(--text-opacity));
  font-weight: 600;
  font-size: 1.25rem;
  -webkit-letter-spacing: -0.025em;
  -moz-letter-spacing: -0.025em;
  -ms-letter-spacing: -0.025em;
  letter-spacing: -0.025em;
}

.emotion-6 > div > nav > .toggleButton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:1024px) {
  .emotion-6 > div > nav > .toggleButton {
    display: none;
  }
}

.emotion-6 > div > nav > .menuList {
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: none;
}

@media (min-width:1024px) {
  .emotion-6 > div > nav > .menuList {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: auto;
  }
}

@media (min-width:1024px) {
  .emotion-6 > div > nav > .menuList {
    display: block;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  border-width: 0;
  border-radius: 0.25rem;
  background-color: transparent;
}

.emotion-2 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-2 svg path {
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  -webkit-transition-duration: 100ms;
  transition-duration: 100ms;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,1,1);
  transition-timing-function: cubic-bezier(0.4,0,1,1);
}

.emotion-2 svg path.open {
  opacity: 0;
}

.emotion-2 svg path.closed {
  opacity: 1;
}

.emotion-4 {
  font-size: 0.875rem;
}

@media (min-width:1024px) {
  .emotion-4 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

.emotion-4 > a {
  display: block;
  margin-top: 1rem;
  --text-opacity: 1;
  color: rgba(160,174,192,var(--text-opacity));
}

@media (min-width:1024px) {
  .emotion-4 > a {
    display: inline-block;
    margin-top: 0;
  }
}

.emotion-4 > a:not(:last-child) {
  margin-right: 1rem;
}

.emotion-8 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width:640px) {
  .emotion-8 > div {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:1024px) {
  .emotion-8 > div {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

.emotion-8 > div > p {
  font-size: 0.875rem;
  --text-opacity: 1;
  color: rgba(160,174,192,var(--text-opacity));
}

.emotion-8 > div > p > a {
  --text-opacity: 1;
  color: rgba(113,128,150,var(--text-opacity));
  margin-left: 0.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="emotion-10 emotion-11"
>
  <div
    className="cookieNotification emotion-0 emotion-1"
    style={
      Object {
        "top": "0",
      }
    }
  >
    <div
      className=""
      style={Object {}}
    >
      This website uses cookies to enhance the user experience.
    </div>
    <div
      className=""
    >
      <button
        aria-label="Accept cookies"
        className="acceptButton"
        id="rcc-confirm-button"
        onClick={[Function]}
        style={Object {}}
      >
        I understand
      </button>
    </div>
  </div>
  <header
    className="emotion-6 emotion-7"
  >
    <div>
      <nav>
        <div
          className="title"
        >
          <span>
            *WEBSITE TITLE*
          </span>
        </div>
        <div
          className="toggleButton"
        >
          <button
            className="emotion-2 emotion-3"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="open"
                d="M6 18L18 6M6 6l12 12"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
              <path
                className="closed"
                d="M4 6h16M4 12h16M4 18h16"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
        </div>
        <div
          className="menuList"
        >
          <div
            className="emotion-4 emotion-5"
          >
            <a
              href="/"
            >
              Docs
            </a>
            <a
              href="/"
            >
              Examples
            </a>
            <a
              href="/"
            >
              Blog
            </a>
          </div>
        </div>
      </nav>
    </div>
  </header>
  <main>
    <a
      href="/"
    >
      link
    </a>
  </main>
  <footer
    className="emotion-8 emotion-9"
  >
    <div>
      <p>
        ©
        2020
         
        *WEBSITE TITLE* —
        <a
          href="https://twitter.com/uitspitss"
          rel="noopener noreferrer"
          target="_blank"
        >
          @uitspitss
        </a>
      </p>
    </div>
  </footer>
</div>
`;
